homeassistant/components/homekit/util.py

```diff
@@ -414,7 +414,11 @@ def cleanup_name_for_homekit(name: str | None) -> str:
     # likely isn't a problem
     if name is None:
         return "None"  # None crashes apple watches
-    return name.translate(HOMEKIT_CHAR_TRANSLATIONS)[:MAX_NAME_LENGTH]
+    return (
+        name.translate(HOMEKIT_CHAR_TRANSLATIONS)
+        .lstrip("-_")[:MAX_NAME_LENGTH]
```

Comment:
Let's make -_ a constant as well to explain the intent

Reply:
Added constant `INVALID_END_CHARS`. Should I put a comment stating something like "Characters that aren't allowed by HomeKit at the two ends post translation" or will it be obvious?

Reply:
I think the name is enough